sfge/src/main/java/com/salesforce/rules/ThreadableRuleExecutor.java

```diff
@@ -182,21 +179,29 @@ public void run() {
                             new Violation.TimeoutViolation(
                                     "Path evaluation timed out after " + TIMEOUT + " ms",
                                     submission.getPathEntry()));
+                    if (LOGGER.isErrorEnabled()) {
+                        LOGGER.error("Timeout Error executing rule. submission=" + submission, ex);
+                    }
                 } else {
                     // If the thread threw another exception, wrap it in a violation, including the
-                    // last line that executed
+                    // last 3 frames that got executed.
+                    String frameString =
+                            Arrays.stream(ex.getStackTrace())
+                                    .limit(3)
+                                    .map(StackTraceElement::toString)
+                                    .collect(Collectors.joining(";"));
```

Comment:
Ordinarily, I'd join stack frames with a newline character. But since the release is coming up soon, I don't want to do anything that risks breaking formatting, so it uses a semi-colon for now. We can change it if we need to.